client/js/src/helpers/data.ts

```diff
@@ -187,14 +187,15 @@ export function handle_payload(
 
 	let updated_payload: unknown[] = [];
 	let payload_index = 0;
-	for (let i = 0; i < dependency.inputs.length; i++) {
-		const input_id = dependency.inputs[i];
+	const deps = type === "input" ? dependency.inputs : dependency.outputs;
```

Comment:
We just need to check which dependencies we care about here and set it accordingly, when handling `inputs` we use `dependency.inputs` when handling `outputs` we use `dependency.outputs`.